### PR TITLE
Follow up from #1359

### PR DESCRIPTION
**Pull request by olafurpg**
Partially addressed the checkboxes in https://github.com/scalameta/scalameta/issues/1362.

* remove global mutable scope
* skip anonymous classes
* merge inner classes with outer classes
* add p.STATIC
* run javacp against JDK in tests

### CODE REVIEW

File: semanticdb/integration/src/main/java/com/javacp/Javacp.java [hunk 1]
```diff
@@ -59,4 +59,7 @@ void packagePrivateMethod() { }
 
     public Serializable anonymous = new Serializable() { };
 
+    static void staticMethod() {}
+    static class StaticClass {}
```
Comment (xeno-by) [inline semanticdb/integration/src/main/java/com/javacp/Javacp.java]: Let's test a static field too, if you don't mind.

File: tests/jvm/src/test/resources/metacp.expect [hunk 2]
```diff
@@ -36,73 +36,6 @@ _empty_.B#`<init>`(). => primaryctor <init>: (): B
 _empty_.B#a(). => def a: : A
   A => _empty_.A#
 
-com/javacp/ClassSuffix$Inner$Bar$Fuz.class
-------------------------------------------
```
Comment (xeno-by) [inline tests/jvm/src/test/resources/metacp.expect]: Nested classes are now stored in their owner's semanticdb files, right?

File: tests/jvm/src/test/resources/metacp.expect [hunk 3]
```diff
@@ -189,13 +94,13 @@ Schema => SemanticDB v3
...
 Language => Java
-Symbols => 56 entries
+Symbols => 66 entries
 
 Symbols:
 _root_. => package _root_
 _root_.com. => package com
 _root_.com.javacp. => package javacp
-_root_.com.javacp.Javacp# => class Javacp[A <: CharSequence with Serializable, B <: Object].{+30 decls}
+_root_.com.javacp.Javacp# => class Javacp[A <: CharSequence with Serializable, B <: Object].{+31 decls}
```
Comment (xeno-by) [inline tests/jvm/src/test/resources/metacp.expect]: Why is it just +1 decl when you added both a static class and a static method?

File: tests/jvm/src/test/resources/metacp.expect [hunk 4]
```diff
@@ -121,65 +54,37 @@ _root_.com.javacp.ClassSuffix# => class ClassSuffix.{+2 decls}
...
-Symbols => 4 entries
-
-Symbols:
-_root_.com.javacp.Javacp#1# => private[javacp] class 1.{+2 decls}
+_root_.com.javacp.ClassSuffix#Inner# => private[javacp] class Inner[A <: Object].{+1 decls}
   extends Object
-  extends Serializable
-_root_.com.javacp.Javacp#1#<init>(Javacp). => private[javacp] def <init>: (arg0: Javacp): Unit
-  arg0 => _root_.com.javacp.Javacp#1#<init>(Javacp).(arg0)
+_root_.com.javacp.ClassSuffix#Inner#<init>(ClassSuffix). => private[javacp] def <init>: (arg0: ClassSuffix): Unit
```
Comment (xeno-by) [inline tests/jvm/src/test/resources/metacp.expect]: I like that `this$N` outer fields no longer show up in SemanticDB. Can we also filter out the corresponding outer parameters from inner class constructors? The heuristic would be "if the class contains a field such that `field.name.startsWith("this$")`, then drop the first parameters of the constructor".

File: semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/Main.scala [hunk 5]
```diff
@@ -128,7 +126,10 @@ object Main {
           files
             .iterator()
             .asScala
-            .filter(f => Files.isRegularFile(f))
+            .filter { f =>
+              !isVisited(f) &&
+              Files.isRegularFile(f)
+            }
```
Comment (xeno-by) [inline semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/Main.scala]: Do we still need sorting? If not, let's go back to having all logic in `visitFile`.

File: tests/jvm/src/test/resources/metacp.expect (hunk 2 above)
Comment (olafurpg) [reply tests/jvm/src/test/resources/metacp.expect]: Correct, these entries are included below see https://github.com/scalameta/scalameta/pull/1367/files#diff-026fd03a8437acdb4213e4e80bf077d1R71

File: tests/jvm/src/test/resources/metacp.expect (hunk 4 above)
Comment (olafurpg) [reply tests/jvm/src/test/resources/metacp.expect]: Done.

File: tests/jvm/src/test/resources/metacp.expect (hunk 3 above)
Comment (olafurpg) [reply tests/jvm/src/test/resources/metacp.expect]: Good catch, I've updated the logic to include direct inner classes as ClassInfo.declarations.

**Pull request merged**